master/addexptrigger.sql

```diff
@@ -0,0 +1,150 @@
+ALTER TABLE experiments ADD COLUMN IF NOT EXISTS seq bigint;
+CREATE SEQUENCE IF NOT EXISTS stream_experiment_seq START 1;
+
+-- trigger function to update sequence number on row modification
+-- this should fire BEFORE so that it can modify NEW directly.
+CREATE OR REPLACE FUNCTION stream_experiment_seq_modify() RETURNS TRIGGER AS $$
+BEGIN
+    NEW.seq = nextval('stream_experiment_seq');
+RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+--
+DROP TRIGGER IF EXISTS stream_experiment_trigger_seq ON experiments;
+CREATE TRIGGER stream_experiment_trigger_seq
+    BEFORE INSERT OR UPDATE OF
+    state, archived, progress, start_time, end_time, notes
+                     ON experiments
+                         FOR EACH ROW EXECUTE PROCEDURE stream_experiment_seq_modify();
+
+-- helper function to create exp jsonb object for streaming
+-- halp!
```

Comment:
halp? 🤣

Reply:
I'm...actually not sure where that came from 😂